### ISSUE TITLE
Remove Analytics tag permission related components and infra

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#4702]
## Feature Description

The biggest change for Analytics as part of the simplification of existing tags is the removal of behavior based on a user's access to a detected existing tag. Accordingly all such behavior and infrastructure related to checking access to an existing tag (property) can be removed.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All components, hooks, selectors, actions, utilities, tests, module REST endpoints, and conditions related to **tag permission** for the Analytics module should be removed

**The PR for this issue should target the [`feature/existing-tag-simplification`](https://github.com/google/site-kit-wp/tree/feature/existing-tag-simplification) branch**

## Implementation Brief

**For the Analytics module only**

* Remove all use of `hasExistingTagPermission` and `hasTagPermission` and the use of `ExistingTagError` and `ExistingGTMPropertyError` components in components, removing any conditions that would render an `ExistingTagError` or `ExistingGTMPropertyError` entirely
	* Also render `StoreErrorNotices` in `SettingsView` unconditionally
	* Also remove the `ExistingTagError`, `ExistingGTMPropertyError` component files
* Remove use of `hasExistingTagPermission` and `hasTagPermission` from `validateCanSubmitChanges`, removing these checks entirely
* Remove logic around existing tags from `getAccounts` resolver entirely, i.e. no longer pass the two params `existingPropertyID` and `existingAccountID`, and remove support for those two parameters from the `GET:accounts-properties-profiles` and `GET:properties-profiles` REST endpoints as well
* Remove all use of `getTagPermission` from `useExistingTagEffect` and drastically simplify it so that all that it does going forward is disable the plugin snippet if the existing tag is the same as the currently configured property ID
* Remove `hasExistingTagPermission`, `hasTagPermission`, and `getTagPermission` store selectors, also the related fixtures
* Remove the `GET:tag-permission` REST endpoint (including the `has_access_to_property` method only used there)

**Storybook**

* Update Analytics setup and settings stories related to existing tags specific to a tag "without access" (i.e. without permissions) to properly reflect the correct non-blocking state
	* Since there will no longer be a concept of an existing tag with/without access these stories may be removed if duplicates of existing stories

### Test Coverage

* Related test coverage should be removed and updated accordingly
* Update `tests/e2e/specs/modules/analytics/setup-with-account-with-tag.test.js` to no longer expect the user to be blocked by an existing tag the user does not have access to, as well as anything else related to tag permissions

## QA Brief

* Test Analytics setup for the following scenarios:
  * new setup: User with existing Analytics properties
  * new setup: User without an Analytics account
  * new setup: Tag on site that does not belong to user
  * new setup: Tag on site that does belong to user
  * edit settings: change settings with a tag that does not belong to the user now on site
  * edit settings: change settings with a tag that does belong to the user now on site
  * edit settings: change settings with tag removed from site

In all scenarios, the user should be able to proceed with Analytics setup and select any account. In any scenario where the existing tag on the site matches the selected Property ID in the Analytics setup, the Analytics snippet should not be placed (should be disabled). In any scenario where the existing tag does NOT match the selected Property ID, the snippet should be placed (enabled).

In the setup this will happen without the toggle appearing; in the settings you should see the toggle go from enabled/disabled based on whether the existing tag matches the selected property ID.

## Changelog entry

* Remove functionality related to checking for existing Analytics tag permission.

**Comment by felixarntz** [comment]
@aaemnnosttv IB almost ✅ , just can you please add at least one note to also remove/adjust any related Storybook stories? Specifically the ones with tag permission errors should no longer be relevant.

**Comment by felixarntz** [comment]
@eclarke1 @FlicHollis @aaemnnosttv Just leaving a note here that whenever this goes into a sprint/release, we should make sure it eventually lands in the same release as #4703, to not cause some weird "intermediate" user experience.

**Comment by felixarntz** [comment]
IB ✅

**Comment by mohitwp** [comment]
@tofumatt 

### QA Update ⚠️ 

In GTM we can connect Google Tag managers tag with Analytics UA tag using Tracking ID. 

![image](https://user-images.githubusercontent.com/94359491/159220394-84d24ba0-13f6-41ef-9112-47ab3f53f869.png)

Similarly, we can connected GTM tag with Analytics GA4 tag using measurement ID.

![image](https://user-images.githubusercontent.com/94359491/159220523-4e1e3726-53b8-47ba-8dfb-c3aa41147332.png)


**Steps;**
1) Create analytics A/c using GA4 tag for Alpha website.
2) Go to GTM.
3) Create GTM manager a/c for Alpha website.
4) Create new tag and connect tag with analytics using Measurement ID of GA4 tag.
5) Copy snippet and paste it in theme header.php file.

1. While testing this PR I found that  if Tag manager container contains either only UA tag or both UA and GA4 tag, then Tag manager automatically detects and shows options to configure analytics.

![image](https://user-images.githubusercontent.com/94359491/159220796-74a10832-5d51-46ef-b313-dc4e724c6f4d.png)

![image](https://user-images.githubusercontent.com/94359491/159220820-da2cedd6-8339-4a56-963e-2e72a523bece.png)

2. If tag manager container contains only GA4 tag then tag manager not shows option to configure analytics and also analytics do not detect any tag.

![image](https://user-images.githubusercontent.com/94359491/159221395-e32f5dec-ad7a-4ea2-b2c9-dfc0c44dcf2c.png)

**Question** : Is this expected behavior that in case of GA4 analytics will not detect any tag ?

cc @wpdarren

**Comment by tofumatt** [comment]
@mohitwp Yes, this issue doesn't deal with GA4 tags, only UA tags, so that is expected. 👍🏻

**Comment by aaemnnosttv** [comment]
@mohitwp this issue is worded a bit oddly, but the detection of a GA4 tag in a GTM container should be handled in https://github.com/google/site-kit-wp/issues/4103 which will be done after the main tag simplification issues 👍

**Comment by mohitwp** [comment]
### QA Update ✅ 

**Verified**

- new setup: User with existing Analytics properties
   The user is able to proceed with Analytics setup and select any account. 


https://user-images.githubusercontent.com/94359491/165042231-53d1f331-65c9-46fd-b66e-68bdeb436c28.mp4



- new setup: User without an Analytics account
Verified :  The user is able to proceed with Analytics setup and select any account

- new setup: Tag on site that does not belong to user
![image](https://user-images.githubusercontent.com/94359491/165040193-fa863da5-3d9d-4dfb-8cb6-8b033e6b1d03.png)

- new setup: Tag on site that does belong to user
Verified: User is able to select account and container that belongs to user.
![image](https://user-images.githubusercontent.com/94359491/165039995-eecac0e1-2c1d-476d-9e22-297aa22eeb3a.png)

- edit settings: change settings with a tag that does not belong to the user now on site
Verified : User is able to select account and container that belongs to user.

edit settings: change settings with a tag that does belong to the user now on site
Verified : User is able to select same tag account and container.

edit settings: change settings with tag removed from site.
Verified : User is able to select tag account and container.

-  Verified :  In any scenario where the existing tag on the site matches the selected Property ID in the Analytics setup, the Analytics snippet not getting place ( disabled). In any scenario where the existing tag does NOT match the selected Property ID, the snippet getting place (enabled).

- Verified : In all scenarios, the user is able to proceed with Analytics setup and select any account.
